### PR TITLE
feat: Introduced meta to action

### PR DESCRIPTION
**Pull request by semoal**
### Todo
- [x] Add to documentation

### CODE REVIEW

**Comment by semoal** [comment]
Give it a look to types @tianzhich if they look good to you, we're ready to :shipit:

File: packages/core/test/dispatcher.test.ts [hunk 1]
```diff
@@ -117,9 +117,47 @@ describe('dispatch:', () => {
...
+						}
+					},
+				},
+			}
+
+			const store = init({
+				models: { count },
+			})
+
+			// TODO: type of `add` inferred incorrectly.
```
Comment (tianzhich) [inline packages/core/test/dispatcher.test.ts:150]: There is an issue(maybe) with [typescript](https://github.com/microsoft/TypeScript/issues/42072). So the optional `meta` isn't supported. You can also click [playground](https://www.typescriptlang.org/play?ts=4.2.0-dev.20201221#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgQWxgQiQB4AVABUwE8AbCTAEzgF45MlaAaOCgLLAYmdp24A+OAG8AUHAVwwdRiwD8ALn40GTZvMUBbYZk38hI2QF9ZsmLTB5qKvWKQBXQwCM0cAD5w7kjMwOjIwPr2juYmYgBKwNjQzGQAzjBQyADmfFy0UgFBIWFIEbZReAAiCKnKMNgAFmgAYtAAQphQlDqqrBwAbhAIzHyCsQNDzFIcANrOuiyjFpgAunCgqMGpcDODw3x7zCsGCmpwABTnAJTsUoTEpGSHB5MSNwBkMoipAKLo6IkYFosPRUngbIo4FoxiJ1iBNsxtocTnAzkVQuFWBtgFttC4WCizpdlAtmGZ5r0bmw7kQSOQKXoXsM3nBPnJIZCan8AURgZhQcAURDFFpifjmNCenoqTSHvSpYs4IcWWyUZzfv9AXyBUKUVoZuiShE+IbMWtsbi5gqRjERMdIUTziTeuTrXxjCJycsZQRaY8GYqYZgVTI1YouZreXAQWDdZCtKbSlj4TjEbbMISLk7xZLxe6TF6TD77nTunn0yH2RyFBGeUDo-zY5DhQpRdnSbnSfmRNDvbdfXKy12Kx9Q9Waxq69qm4obLIAPQAKkX8kXcAqcA6UDEl3bLs7lP7Jf9bqVr1HVYn3K1DYFAG5rDcArvnXoD4y4B7MFoEkkoCl0kyJAcnEfJiz9eVy1-ZIyAAOngiRKxReDYIfKwrlXec7AcPAtzEapakweomigVooC3IdeiWEwJFsehhDgLxOi0LcHwXZdVzgej4CYqBRVfFh30VL8f0SGDAOyXJJHAwcAxtZV-AHUs5L4aD-zSDJJNAxCUQANU6BBMC8ei4AAcl40zvkCMFWB8dBoDwHxsk4VJUgQLIk1g84ACYABYAFZfIwyEfnAQEIjgbzOCgLJPBxGBUj4Lx3HgLIIHgAAGLzvP8wLgvDJAwBS2CEpAHyAHY+AATiuLR8CQaLYuMFBo2gMyBOYSyAHdMG2JB0qUKAIH6YYIlgzDZF4642KXFc4DXDcOgALx3Pc9Fdcsv0LEQZOU08gyQ9VryjGNgDQibFswFbZhU9NzRTS1njPYZ7UUR1dseJ7lQvb5jvrU64BbKF0zhBEkUmTNEwii00zkzMxVJDbSQ+yDh2+1kx3HWsb1OuMRSzDqhOYFHKI-dHVXHX7I3+xtBWbPUdihm0mfusGdluoNXtOAnxSRqjPwLXsiyPCDScDZZDqxyccdpvHW0CYIMSTUHU22IN4bWwS8WHLahZ2kXZP2iWLzDK9qenOnZwZhHeiJ7tv3TEmOeNjHLw5bGTtl+nZG4xirpYq62KYpbpsmq7zgARiuB9g8jvhpDM4PTK0CPAej2QgA) for more details.

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -359,11 +376,30 @@ export type ExtractRematchDispatcherFromReducer<
  * Otherwise, it describes dispatcher which accepts one argument (payload)
  * and returns an action.
  */
-export type RematchDispatcher<TPayload = void> = [TPayload] extends [void]
-	? (() => Action<void>) & { isEffect: false }
-	: undefined extends TPayload
-	? ((payload?: TPayload) => Action<TPayload>) & { isEffect: false }
-	: ((payload: TPayload) => Action<TPayload>) & { isEffect: false }
+export type RematchDispatcher<TPayload = void, TMeta = void> = [
```
Comment (tianzhich) [inline packages/core/src/types.ts:379]: I added the type support for optional `meta` in `RematchDispatcher` and `ExtractRematchDispatcherFromReducer`.

**Review by tianzhich — APPROVED** [review]

File: packages/core/test/dispatcher.test.ts (hunk 1 above)
Comment (tianzhich) [reply packages/core/test/dispatcher.test.ts]: I fixed that.

**Comment by tianzhich** [comment]
I've checked the types. If you have nothing to complete we can merge this.

**Pull request merged**